### PR TITLE
fix: [CDS-82203]: Pipeline execution shows success post abort

### PR DESCRIPTION
**Pull request by Aninditaa-ch**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54979)

### CODE REVIEW

**Comment by Aninditaa-ch** [comment]
trigger uts

File: pipeline-service/modules/orchestration/src/test/java/io/harness/engine/interrupts/handlers/AbortAllInterruptHandlerTest.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.anyString;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
```
Comment (ankittiwari-harness) [inline pipeline-service/modules/orchestration/src/test/java/io/harness/engine/interrupts/handlers/AbortAllInterruptHandlerTest.java:20]: nit: remove *

**Comment by Aninditaa-ch** [comment]
trigger uts

File: pipeline-service/modules/orchestration/src/test/java/io/harness/engine/interrupts/handlers/AbortAllInterruptHandlerTest.java (hunk 1 above)
Comment (Aninditaa-ch) [reply pipeline-service/modules/orchestration/src/test/java/io/harness/engine/interrupts/handlers/AbortAllInterruptHandlerTest.java]: done

**Comment by Aninditaa-ch** [comment]
trigger SmartPRChecks

**Comment by Aninditaa-ch** [comment]
trigger smartchecks

**Comment by Aninditaa-ch** [comment]
trigger messagecheck

**Comment by Aninditaa-ch** [comment]
trigger gitleaks

**Comment by Aninditaa-ch** [comment]
trigger messagecheck

**Comment by Aninditaa-ch** [comment]
trigger messagecheck

**Review by sahilhindwani-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54979)* status: Some of the checks are too old. 2 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/interrupts/handlers/AbortAllInterruptHandler.java` line 61 at r2](https://reviewable.io/reviews/harness/harness-core/54979#-NiZoUBK63wM5_jeTUR5:-NiZoUBK63wM5_jeTUR6:b-65coyh) ([raw file](https://github.com/harness/harness-core/blob/aa8251a423845cdb318c8298dd4b1dbb69ff9ecc/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/interrupts/handlers/AbortAllInterruptHandler.java#L61)):*
> ```Java
>   public Interrupt registerInterrupt(Interrupt interrupt) {
>     Interrupt savedInterrupt = validateAndSave(interrupt);
>     Status status = planExecutionService.getStatus(interrupt.getPlanExecutionId());
> ```

Can we add this condition in handleInterrupt itself?
There we are already making the same call.

**Review by shalini-agr — APPROVED** [review]

**Review by Aninditaa-ch — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54979)* status: Some of the checks are too old. 2 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/interrupts/handlers/AbortAllInterruptHandler.java` line 61 at r2](https://reviewable.io/reviews/harness/harness-core/54979#-NiZoUBK63wM5_jeTUR5:-NicOfrG4hhYvWmhjgzD:b-j3mna2) ([raw file](https://github.com/harness/harness-core/blob/aa8251a423845cdb318c8298dd4b1dbb69ff9ecc/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/interrupts/handlers/AbortAllInterruptHandler.java#L61)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

Can we add this condition in handleInterrupt itself?
There we are already making the same call.
</blockquote></details>

The call is made in validateAndSave not in HandleInterrupt.So this cannot be added in that right?

**Comment by Aninditaa-ch** [comment]
trigger uts

**Review by shalini-agr — APPROVED** [review]

**Comment by Aninditaa-ch** [comment]
trigger ut3

**Comment by Aninditaa-ch** [comment]
trigger reviewCheck

**Pull request merged**